src/runtime-rs/crates/hypervisor/src/device/driver/protection_device.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2025 Red Hat
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::{
+    device::{topology::PCIeTopology, Device, DeviceType},
+    Hypervisor as hypervisor,
+};
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+
+#[derive(Debug, Clone)]
+pub enum ProtectionDeviceConfig {
+    SevSnp(SevSnpConfig),
```

Comment:
seems like a reasonable approach for TDX too. 

current parameters include:
`'{"qom-type":"tdx-guest","id":"tdx","attributes": <u64>, "mrconfigid":"<string>","mrowner":"<string>","mrownerconfig":"<string>","quote-generation-socket":{"type": "vsock", "cid":"2","port":"4050"}}'`

where `mrconfigid` is going be used similarly than `host_data`.

The Go runtime as `port` configurable atm.

Reply:
Yes, this actually is the main goal of this PR - to add a framework which all TEE support can be reasonably easily plugged into.  I imagine `Tdx` will be the next member of this `enum` to be added.

Reply:
I tried to hint that perhaps some of those basic TDX settings could already be included in this PR ;-)

Reply:
I'd really prefer not to overload this PR any more.  I promise though all necessary cooperation when TDX support is added, hopefully soon. :-)